esmvalcore/_recipe.py

```diff
@@ -1061,6 +1062,12 @@ def expand_ensemble(variable):
 
         return expanded
 
+    def _add_project_variable_mappings(self, variable):
+        mappings = get_variable_mappings(variable["project"],
+                                         variable["dataset"])
+        mapping = mappings[variable["mip"]][variable["short_name"]]
```

Comment:
A mapping dict is not supposed to have keys for all mips, and when it has, not all variables have necessary an entry